File: Patrones Creacionales/Builder/Ejemplo.py
class Car:
    def __init__(self):
        self.parts = []

    def add(self, part):
        self.parts.append(part)

    def show(self):
        return ", ".join(self.parts)

class CarBuilder:
    def __init__(self):
        self.car = Car()

    def add_engine(self):
        self.car.add("Motor")
        return self

    def add_wheels(self):
        self.car.add("Ruedas")
        return self

    def build(self):
        return self.car

car = CarBuilder().add_engine().add_wheels().build()
print(car.show())

File: Patrones Creacionales/Builder/Ejemplo.js
class Car {
  constructor() { 
    this.parts = []; 
  }

  add(part) { 
    this.parts.push(part); 
  }

  show() { 
    return this.parts.join(", "); 
  }
}

class CarBuilder {
  constructor() { 
    this.car = new Car(); 
  }

  addEngine() {
    this.car.add("Motor");
    return this;
  }

  addWheels() {
    this.car.add("Ruedas");
    return this;
  }

  build() { 
    return this.car; 
  }
}

const car = new CarBuilder().addEngine().addWheels().build();
console.log(car.show());